### DISCUSSION TITLE
Implementation errors when run with mypy

### DISCUSSION THREAD

**Discussion by opossalite** [PyO3/maturin#1525] [Q&A]
### Bug Description

After providing full type hinting following the guide [here](https://github.com/PyO3/maturin/blob/084cfaced651b28616aeea1f818bdc933a536bfe/guide/src/project_layout.md), specifically right under the "Mixed Rust/Python project" section, I'm able to see type hinting in vim using pyright. However when I run mypy on the same file, it returns this error: 
```
main.py:1: error: Cannot find implementation or library stub for module named "tools"  [import]
```

### Your maturin version (`maturin --version`)

0.14.13

### Your Python version (`python -V`)

3.10.9

### Your pip version (`pip -V`)

23.0.1

### What bindings you're using

pyo3

### Does `cargo build` work?

- [X] Yes, it works

### If on windows, have you checked that you aren't accidentally using unix path (those with the forward slash `/`)?

- [X] Yes

### Steps to Reproduce

I am running Arch Linux.

My main.py:
```
import tools

print(tools.sum_as_string(1, 3))
```

My maturin code is the init code. The project is called "tools", I have a subdirectory called "tools" in the root, and inside is a __init__.py with this:
```
"""Tools implemented in Rust to speed up computation."""
from .tools import *

__doc__ = .tools.__doc__
```
and a py.typed empty file, and finally a tools.pyi file for type hinting. This results in perfect type hinting in vim with pyright, however mypy returns the error shown above. Perhaps I've simply missed something or formatted my directories wrong?

**Comment by messense** [comment]
> however mypy returns the error shown above. Perhaps I've simply missed something or formatted my directories wrong?

I don't know, you probably should check how `mypy` handles such type stubs?
